bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -183,11 +184,42 @@ private String getFirstLocalIPv4Address() {
         return broadcastAddresses;
     }
 
+    @Override
+    public String getPrimaryBroadcastAddress() {
```

Comment:
Could you please some test cases for this method in https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/net/NetUtilTest.java ?

`byte` magic is always tricky to debug and easy to break, so I think it would be good to have some test cases for that.

Reply:
I will look into this. I have not done test cases so far, so need to get over the learning curve.

Reply:
I added test cases. I split out the main calculation logic in a static method, so the logic can be tested more easily.